software/base/src/main/java/org/apache/brooklyn/entity/software/base/InboundPortsUtils.java

```diff
@@ -77,12 +82,17 @@
             Set<ConfigKey<?>> configKeys = Sets.newHashSet(extraConfigKeys);
             configKeys.addAll(entity.getEntityType().getConfigKeys());
             // Also add dynamically added config keys
-            configKeys.addAll(((EntityInternal)entity).config().getBag().getAllConfigAsConfigKeyMap().keySet());
+            Map<ConfigKey<?>, ?> configuredConfigKeys = ImmutableMap.copyOf(((EntityInternal) entity).config().getBag().getAllConfigAsConfigKeyMap());
+            configKeys.addAll(configuredConfigKeys.keySet());
+
+            if (portRegex == null) {
+                portRegex = ".*\\.port"; // defaults to legacy regex if not specified
+            }
```

Comment:
please avoid whitespace changes, import reorgs (I know you know this now!), and preference style changes like the above (if it's an egregiously bad style change it)

those things all make PR's harder to review